fixtures/gists-app/app/routes/resources/settings.tsx

```diff
@@ -36,12 +36,13 @@ function reloadCss() {
 export default function Settings() {
   let settings = useLoaderData<Record<string, string>>();
   let { state, submission } = useTransition();
+  let isSubmission = Boolean(submission);
 
   useEffect(() => {
-    if (state === "loading" && submission) {
+    if (state === "loading" && isSubmission) {
       reloadCss();
     }
-  }, [state]);
+  }, [state, isSubmission]);
```

Comment:
This was a "exhaustive-deps" warning. Pretty sure I got the spirit of this code while maintaining the rule, but let me know if we need to use a ref to workaround it instead.